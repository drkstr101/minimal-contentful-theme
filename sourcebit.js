module.exports = {
    plugins: [
        {
            module: require('sourcebit-source-contentful'),
            options: {
                accessToken: process.env['CONTENTFUL_ACCESS_TOKEN'],
                environment: 'master',
                spaceId: 'krtnabbrkhtd'
            }
        },
        {
            module: require('sourcebit-target-next'),
            options: {
                pages: function (objects, utils) {
                    return objects.reduce((pages, page) => {
                        if (page.__metadata.modelName === 'Page' && page.__metadata.source === 'sourcebit-source-contentful') {
                            return pages.concat({
                                path: '{slug}',
                                page
                            });
                        }

                        return pages;
                    }, []);
                },
                commonProps: function (objects, utils) {
                    return {
                        site: objects.find((object) => object.__metadata.modelName === 'SiteConfig')
                    };
                }
            }
        }
    ]
};
