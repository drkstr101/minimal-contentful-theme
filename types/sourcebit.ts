// THIS FILE IS AUTOMATICALLY GENERATED. DO NOT MODIFY IT.

import { Asset, Entry } from 'contentful';
import { Document } from '@contentful/rich-text-types';

export interface Metadata {
    __metadata: {
        id: string;
        modelName: string;
        urlPath?: string;
    };
}

export interface IButton extends Metadata {
    /** Label */
    label: string;

    /** URL */
    url?: string | undefined;

    /** Color scheme */
    theme: 'primary' | 'secondary';
}


export interface ICard extends Metadata {
    /** Heading */
    heading: string;

    /** Subheading */
    subheading?: string | undefined;

    /** URL */
    url?: string | undefined;
}


export interface ICardGridSection extends Metadata {
    /** Heading */
    heading: string;

    /** Subheading */
    subheading?: string | undefined;

    /** Cards */
    cards: ICard[];
}


export interface IFooterConfig extends Metadata {
    /** Body */
    body: string;
}


export interface IHeroSection extends Metadata {
    /** Heading */
    heading?: string | undefined;

    /** Subheading */
    subheading?: string | undefined;

    /** Buttons */
    buttons: IButton[];
}


export interface IPage extends Metadata {
    /** Slug */
    slug: string;

    /** Layout */
    layout: string;

    /** Title */
    title: string;

    /** Sections */
    sections: (ICardGridSection | IHeroSection)[];
}


export interface ISiteConfig extends Metadata {
    /** Title */
    title: string;

    /** Footer */
    footer: IFooterConfig;
}


export type CONTENT_TYPE = 'Button' | 'Card' | 'CardGridSection' | 'FooterConfig' | 'HeroSection' | 'Page' | 'SiteConfig';

export type LOCALE_CODE = 'en-US';

export type CONTENTFUL_DEFAULT_LOCALE_CODE = 'en-US';
