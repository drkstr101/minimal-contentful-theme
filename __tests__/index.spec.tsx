import { render, screen } from '@testing-library/react';
import { FlexiblePage, getStaticProps, getStaticPaths } from '../pages/[[...slug]]';
import sourcebit from 'sourcebit';
import config from '../sourcebit.js';
import { act } from 'react-dom/test-utils';
import { omit, transform, isObject, isArray, keyBy, isNil } from 'lodash';

const sourcebitOptions = {};

function deepOmit(input, props) {
    function omitDeepOnOwnProps(obj) {
        if (typeof input === 'undefined') {
            return input;
        }

        if (!Array.isArray(obj) && !isObject(obj)) {
            return obj;
        }

        if (Array.isArray(obj)) {
            return deepOmit(obj, props);
        }

        const o = {};
        for (const [key, value] of Object.entries(obj)) {
            o[key] = !isNil(value) ? deepOmit(value, props) : value;
        }

        return omit(o, props);
    }

    if (arguments.length > 2) {
        props = Array.prototype.slice.call(arguments).slice(1);
    }

    if (Array.isArray(input)) {
        return input.map(omitDeepOnOwnProps);
    }

    return omitDeepOnOwnProps(input);
}

describe('minimal-contentful-theme', () => {
    // ensure sourcebit has loaded
    beforeEach(async () => {
        console.log('Loading source data...');
        const data = await sourcebit.fetch(config, sourcebitOptions);
        // console.log(data);
    });

    it('getStaticPaths', async () => {
        expect(await getStaticPaths()).toEqual({
            paths: ['/'],
            fallback: false
        });
    });

    it('getStaticProps', async () => {
        const {
            props: { site, page }
        } = await getStaticProps({ params: { slug: '/' } });

        // TODO use `__metadata.modelName` rather than including a special `type` field
        expect(deepOmit(site, '__metadata')).toEqual({
            type: 'SiteConfig',
            title: 'Stackbit',
            footer: {
                type: 'FooterConfig',
                body: 'Made by [Stackbit](https://www.stackbit.com/)\n'
            }
        });

        expect(deepOmit(page, '__metadata')).toEqual({
            slug: '/',
            layout: 'Page',
            title: 'Stackbit Next.js Starter',
            sections: [
                {
                    type: 'HeroSection',
                    heading: 'Welcome to Stackbit!',
                    subheading: "You've just [unlocked visual editing capabilities](https://www.stackbit.com/) in this Next.js app.",
                    buttons: [
                        { label: 'Start Building', url: 'https://docs.stackbit.com/getting-started/', theme: 'primary' },
                        { label: 'Read the Docs', url: 'https://docs.stackbit.com/', theme: 'secondary' }
                    ]
                },
                {
                    type: 'CardGridSection',
                    heading: 'Jump to Topic',
                    subheading: 'Or jump right to a specific topic to help you build your site.',
                    cards: [
                        {
                            heading: 'How Stackbit Works →',
                            subheading: 'Follow an end-to-end guide to learn the inner-workings of Stackbit.',
                            url: 'https://docs.stackbit.com/conceptual-guides/how-stackbit-works/'
                        },
                        {
                            heading: 'Pages →',
                            subheading: 'Add a new type of page to your site, while touching on content modeling and data retrieval.',
                            url: 'https://docs.stackbit.com/how-to-guides/content/'
                        },
                        {
                            heading: 'Components →',
                            subheading: 'Make components editable, add styles, and provide content presets to speed up content editing.',
                            url: 'https://docs.stackbit.com/how-to-guides/components/'
                        },
                        {
                            heading: 'Styling →',
                            subheading: 'Set up global styles and add a styling toolbar to individual components in the visual editor.',
                            url: 'https://docs.stackbit.com/how-to-guides/styles/'
                        }
                    ]
                }
            ]
        });
    });

    it('FlexiblePage', async () => {
        const { props } = await getStaticProps({ params: { slug: '/' } });
        let container: HTMLElement | null = null;
        await act(() => {
            container = render(<FlexiblePage {...props} />).container;
        });
        expect(container).toBeInstanceOf(HTMLElement);
        expect(container).toMatchSnapshot();

        const heading = screen.getByText('Welcome to Stackbit!');
        expect(heading).toBeInTheDocument();
    });
});
