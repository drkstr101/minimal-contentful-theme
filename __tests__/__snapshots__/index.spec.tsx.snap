// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage without crashing 1`] = `
<div>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Stackbit Next.js Starter
      </h1>
    </main>
  </div>
</div>
`;
